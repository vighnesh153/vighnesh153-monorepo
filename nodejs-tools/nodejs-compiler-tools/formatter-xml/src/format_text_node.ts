import { XmlTextNode } from '@vighnesh153/parser-xml';
import { buildIndentationSpace } from './build_indentation_space.ts';

type FormatTextNodeOptions = {
  textNode: XmlTextNode;
  indentationLevel: number;
  indentation: number;
};

export function formatTextNode({ textNode, indentationLevel, indentation }: FormatTextNodeOptions): string {
  return buildIndentationSpace({ indentationLevel, indentation }) + textNode.text.tokenLiteral.trim();
}
