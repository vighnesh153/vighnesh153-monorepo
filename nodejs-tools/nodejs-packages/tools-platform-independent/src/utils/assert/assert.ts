import { not } from '../not';

export function assert(condition: boolean, message = 'Assertion failed') {
  if (not(condition)) {
    throw new Error(message);
  }
}
