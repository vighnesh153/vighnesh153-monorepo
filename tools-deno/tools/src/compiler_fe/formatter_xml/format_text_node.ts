import type { XmlTextNode } from "@/compiler_fe/parser_xml/mod.ts";
import { buildIndentationSpace } from "./build_indentation_space.ts";

type FormatTextNodeOptions = {
  textNode: XmlTextNode;
  indentationLevel: number;
  indentation: number;
};

export function formatTextNode(
  { textNode, indentationLevel, indentation }: FormatTextNodeOptions,
): string {
  return buildIndentationSpace({ indentationLevel, indentation }) +
    textNode.text.tokenLiteral.trim();
}
