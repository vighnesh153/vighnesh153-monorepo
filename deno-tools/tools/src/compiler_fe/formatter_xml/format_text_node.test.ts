import { assertEquals } from "@std/assert";
import { assertSnapshot } from "@std/testing/snapshot";
import type { XmlTagNode, XmlTextNode } from "@/compiler_fe/parser_xml/mod.ts";
import { formatTextNode } from "./format_text_node.ts";
import { parseProgram } from "./test_utils.ts";

Deno.test("formatTextNode should format text node with 0 indentation level", async (t) => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 4,
      indentationLevel: 0,
    }),
  );
});

Deno.test("formatTextNode should format text node with 2 indentation level", async (t) => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 4,
      indentationLevel: 2,
    }),
  );
});

Deno.test("formatTextNode should format text node with 2 indentation level and 3 indentation", async (t) => {
  const [parser, program] = parseProgram(
    `< pokemon  >  Pikachu    < /  pokemon >`,
  );

  assertEquals(parser.errors.length, 0);
  assertEquals(program.statements.length, 1);
  await assertSnapshot(
    t,
    formatTextNode({
      textNode: (program.statements[0] as XmlTagNode)
        .children[0] as XmlTextNode,
      indentation: 3,
      indentationLevel: 2,
    }),
  );
});
