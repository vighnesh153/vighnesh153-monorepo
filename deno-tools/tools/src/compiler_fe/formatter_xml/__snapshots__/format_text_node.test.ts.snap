export const snapshot = {};

snapshot[`formatTextNode should format text node with 0 indentation level 1`] = `"Pikachu"`;

snapshot[`formatTextNode should format text node with 2 indentation level 1`] = `"        Pikachu"`;

snapshot[`formatTextNode should format text node with 2 indentation level and 3 indentation 1`] = `"      Pikachu"`;
